backend/benefit/applications/api/v1/serializers.py

```diff
@@ -67,21 +67,25 @@
 from users.utils import get_company_from_request, get_request_user_from_context
 
 
-class ApplicationBasisSerializer(serializers.ModelSerializer):
+class DynamicFieldsModelSerializer(serializers.ModelSerializer):
     """
-    Only the unique identifier is exposed through the API.
+    A ModelSerializer that takes an additional `fields` argument that
+    controls which fields should be displayed.
     """
 
-    class Meta:
-        model = ApplicationBasis
-        fields = [
-            "identifier",
-        ]
-        extra_kwargs = {
-            "identifier": {
-                "help_text": "Unique slug that identifies the basis",
-            }
-        }
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if fields := self.context.get("fields"):
+            # Drop any fields that are not specified in `fields`
+            allowed = set(fields)
+            existing = set(self.fields)
+            for field_name in existing - allowed:
+                self.fields.pop(field_name)
+
+        if exclude_fields := self.context.get("exclude_fields"):
+            for field_name in exclude_fields:
+                self.fields.pop(field_name)
```

Comment:
An observation:
 - This might raise a KeyError if field_name in exclude_fields is not present in self.fields at this point.